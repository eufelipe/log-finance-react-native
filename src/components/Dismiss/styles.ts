import styled from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import Colors from 'styles/colors';

export const Container = styled.TouchableOpacity``;

export const DismissIcon = styled(Icon).attrs({
  name: 'chevron-down-circle',
  color: `${Colors.white}50`,
  size: 32,
})``;
