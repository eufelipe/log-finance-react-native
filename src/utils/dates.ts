import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
