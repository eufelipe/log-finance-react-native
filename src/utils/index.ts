export {default as number} from './number';
export {default as strings} from './strings';
