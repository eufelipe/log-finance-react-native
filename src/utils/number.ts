export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
