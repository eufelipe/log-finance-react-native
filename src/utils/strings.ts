export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
