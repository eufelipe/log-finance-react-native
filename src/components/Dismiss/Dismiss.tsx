import React from 'react';
import {useNavigation} from '@react-navigation/core';

import {Container, DismissIcon} from './styles';

const Dismiss = (): JSX.Element => {
  const {goBack} = useNavigation();
  return (
    <Container onPress={goBack}>
      <DismissIcon />
    </Container>
  );
};

export default Dismiss;
