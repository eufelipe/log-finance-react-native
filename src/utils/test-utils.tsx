/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
