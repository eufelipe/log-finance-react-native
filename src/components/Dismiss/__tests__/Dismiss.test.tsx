/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render} from 'utils/test-utils';
import Dismiss from '../Dismiss';

describe('Dismiss', () => {
  it('Test match snapshot Dismiss', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<Dismiss {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
