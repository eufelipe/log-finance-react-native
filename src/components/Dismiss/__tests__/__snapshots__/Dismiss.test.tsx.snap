// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dismiss Test match snapshot Dismiss 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff50",
          "fontSize": 32,
        },
        Array [
          Object {},
        ],
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    з░гд
  </Text>
</View>
`;
